backend/src/gh.rs

```diff
@@ -299,6 +309,134 @@ impl GitHubClient {
         }
     }
 
+    /// Updates an existing pull request on GitHub with the specified details.
+    /// 
+    /// # Arguments
+    /// - `pr_number` - The pull request number to update.
+    /// - `pr_title` - Optional new title for the pull request.
+    /// - `pr_description` - Optional updated description for the pull request.
+    /// - `base_branch` - Optional target base branch to update the pull request against.
+    /// - `issue_numbers` - Optional list of issue numbers to link to the pull request. Each issue
+    ///   will be referenced in the pull request description using the "Closes #<issue_number>" syntax.
+    /// 
+    /// # Returns
+    /// Returns the URL of the updated pull request if the operation is successful.
+    /// 
+    /// # Errors
+    /// Returns an error if:
+    /// - The repository name cannot be determined.
+    /// - The GitHub API request fails, including cases where the response does not contain the expected `html_url` field.
+    /// - Network or deserialization issues occur while processing the response.
+    pub async fn update_pull_request(
+        &self,
+        pr_number: u64,
+        pr_title: Option<&str>,
+        pr_description: Option<&str>,
+        base_branch: Option<&str>,
+        issue_numbers: Option<Vec<u64>>,
```

Comment:
This is a prime candidate for the Builder pattern, see https://docs.rs/bon/latest/bon/. It would enable people to do something like
`client.update_pull_request(43).pr_title("Foo")`, and omit any information that's not necessary for inclusion.

I'll leave it up to you whether you make that change now, or open an issue to make it later.

Reply:
Made an issue will take care of it after the gh-client update.